spec/operatoroverloading.dd

```diff
@@ -283,11 +300,16 @@ if (e)   =>  if (e.opCast!(bool))
 if (!e)  =>  if (!e.opCast!(bool))
 ---
 
-        $(P etc., whenever a bool result is expected. This only happens, however, for
-        instances of structs. Class references are converted to bool by checking to
+        $(P and similarly for `assert(e)` and
+        $(DDSUBLINK spec/expression, logical_expressions, logical operators) used
```

Comment:
I think there are a few more, `static assert`, `while`, `for`, `do ... while`

Reply:
Thanks, changed to use 'other boolean conditional expressions'.